src/analyzer.ts

```diff
@@ -15,8 +16,17 @@ export class Analyzer extends AnalyzerGen {
 
 	constructor(dartVMPath: string, analyzerPath: string) {
 		super();
+
 		console.log(`Starting Dart analysis server...`);
-		this.analyzerProcess = child_process.spawn(dartVMPath, [analyzerPath]);
+
+		let args = [analyzerPath];
+
+		// Optionally start the analyzer's diagnostic web server on the given port.
+		let port = getConfig('analyzerDiagnosticsPort');
+		if (port)
+			args.push(`--port=${port}`);
```

Comment:
hmm, it may not be documented - it's not really a user facing feature though - just used by the analysis server team when things go badly wrong.  It's useful when trying to get enough info to diagnose a problem. Also useful when building an analysis server client - you can see what it thinks the state of the overlays are, the priority sources, ...